src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -0,0 +1,122 @@
+import Boom from 'boom';
+import { get } from 'lodash';
+
+import {
+  createFindQuery,
+  handleEsError,
+} from './lib';
+
+export class SavedObjectsClient {
+  constructor(kibanaIndex, callAdminCluster) {
+    this._kibanaIndex = kibanaIndex;
+    this._callAdminCluster = callAdminCluster;
+  }
+
+  async create(type, body = {}) {
+    const response = await this._withKibanaIndex('index', {
+      type,
+      body
+    });
+
+    return {
+      type: response._type,
+      id: response._id,
+      version: response._version,
+      attributes: body
+    };
+  }
+
+  async delete(type, id) {
+    const response = await this._withKibanaIndex('delete', {
+      type,
+      id,
+      refresh: 'wait_for'
+    });
+
+    if (get(response, 'found') === false) {
+      throw Boom.notFound();
+    }
+
+    return get(response, 'result') === 'deleted';
```

Comment:
I am not sure why _result_ would be anything other than deleted. I have removed the boolean return and we can amend a return later if we find it necessary. For now, it will complete without a return value when it's completed and throw any error, like if the document is not found.